src/camt/053/CashManagementEndOfDayReport.ts

```diff
@@ -36,6 +36,32 @@ export class CashManagementEndOfDayReport {
     this._statements = config.statements;
   }
 
+  /**
+   * Creates and configures the XML Parser
+   *
+   * @returns {XMLParser} A configured instance of XMLParser
+   */
+  private static getParser(): XMLParser {
+    return new XMLParser({
+      ignoreAttributes: false,
+      tagValueProcessor: (
+        tagName,
+        tagValue,
+        _jPath,
+        _hasAttributes,
+        isLeafNode,
+      ) => {
+        /**
+         * Codes and Entry References can look like numbers and get parsed
+         * appropriately. We don't want this to happen, as they contain leading
+         * zeros or are too long and overflow.
+         */
+        if (isLeafNode && ['Cd', 'NtryRef'].includes(tagName)) return undefined;
```

Comment:
Can you explain how this works? Is there a risk that we don't return useful information? 

Or does this basically "skip" empty leaf nodes

Reply:
According to [the docs](https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/docs/v4/2.XMLparseOptions.md#tagvalueprocessor), if this function returns `undefined` then the value is left as a string and not parsed. I added the guard of `isLeafNode` because I only want the values there to stay strings when they don't have nested properties (i.e. `<NtryRef><someVal>1234</someVal></NtryRef>`)

Reply:
Yeah, that's on purpose. The type of the field is defined by `someVal`. If `someVal` dictates that it should be a number then we parse it, otherwise we should add it to the list.